### ISSUE TITLE
twisted.test.test_banana fails on Python 3

### ISSUE THREAD

**Issue by twisted-trac** [twisted/twisted#8667]
|[<img alt="rodrigc's avatar" src="https://avatars.githubusercontent.com/u/1895943?s=50" width="50" height="50">](https://github.com/rodrigc)| @rodrigc reported|
|-|-|
|Trac ID|trac#8667|
|Type|enhancement|
|Created|2016-07-25 03:11:09Z|
|Branch|https://github.com/twisted/twisted/tree/8667-rodrigc-banana-py3|

I tried to run the following:

```
python3 bin/trial twisted/test/test_banana.py
```

and got this:

```
python3 bin/trial twisted/test/test_banana.py
twisted/test/test_banana
  py ...                                                                [ERROR]

===============================================================================
[ERROR]
Traceback (most recent call last):
  File "/Users/crodrigues/twisted2/twisted/trial/runner.py", line 784, in loadByName
    return self.suiteFactory([self.findByName(name, recurse=recurse)])
  File "/Users/crodrigues/twisted2/twisted/trial/runner.py", line 685, in findByName
    return self.loadFile(_name, recurse=recurse)
  File "/Users/crodrigues/twisted2/twisted/trial/runner.py", line 655, in loadFile
    module = SourceFileLoader(name, fileName).load_module()
  File "<frozen importlib._bootstrap_external>", line 383, in _check_name_wrapper
    
  File "<frozen importlib._bootstrap_external>", line 805, in load_module
    
  File "<frozen importlib._bootstrap_external>", line 664, in load_module
    
  File "<frozen importlib._bootstrap>", line 268, in _load_module_shim
    
  File "<frozen importlib._bootstrap>", line 693, in _load
    
  File "<frozen importlib._bootstrap>", line 673, in _load_unlocked
    
  File "<frozen importlib._bootstrap_external>", line 660, in exec_module
    
  File "<frozen importlib._bootstrap>", line 222, in _call_with_frames_removed
    
  File "twisted/test/test_banana.py", line 4, in <module>
    import StringIO
builtins.ImportError: No module named 'StringIO'

twisted/test/test_banana.py
-------------------------------------------------------------------------------
Ran 1 tests in 0.074s

FAILED (errors=1)
```

<details><summary>Searchable metadata</summary>

```
trac-id__8667 8667
type__enhancement enhancement
reporter__rodrigc rodrigc
priority__normal normal
milestone__Python_3_x Python-3.x
branch__8667_rodrigc_banana_py3 8667-rodrigc-banana-py3
branch_author__ 
status__closed closed
resolution__fixed fixed
component__core core
keywords__None None
time__1469416269767581 1469416269767581
changetime__1469431178923145 1469431178923145
version__None None
owner__GitHub__noreply_____ GitHub <noreply@...>

```
</details>

**Comment by twisted-trac** [comment]
|[<img alt="rodrigc's avatar" src="https://avatars.githubusercontent.com/u/1895943?s=50" width="50" height="50">](https://github.com/rodrigc)<a name="note_1"></a>|@rodrigc commented|
|-|-|

See:
https://github.com/twisted/twisted/pull/407

**Comment by twisted-trac** [comment]
|[<img alt="acabhishek942's avatar" src="https://avatars.githubusercontent.com/u/7778186?s=50" width="50" height="50">](https://github.com/acabhishek942)<a name="note_2"></a>|@acabhishek942 commented|
|-|-|

Changes look good. Please merge

**Comment by twisted-trac** [comment]
|<img alt="GitHub's avatar" src="https://avatars.githubusercontent.com/u/0?s=50" width="50" height="50"><a name="note_3"></a>|GitHub set owner to GitHub<br>GitHub set status to `closed`|
|-|-|

In changeset af87497e6bcd6bf1c59293399e15a7fb8997fe5b
```
#!CommitTicketReference repository="" revision="af87497e6bcd6bf1c59293399e15a7fb8997fe5b"
Merge pull request #407 from twisted/8667-rodrigc-banana-py3

Author: rodrigc
Reviewer: acabhishek942
Fixes: #8667

Port twisted.spread.banana to Python 3
```
